src/utils/utils.cpp

```diff
@@ -189,6 +189,17 @@ bool load_rawfile(const std::string& rawfilename, const Eigen::RowVector3i& dims
     return true;
 }
 
+void debug_group_action(const std::string& action, const char* message) {
```

Comment:
Using a string as an action here is problematic. What happens if you pass in `"POPs"` or some other invalid string. Then the function does nothing. Let's split this function into 2 functions:
`push_gl_debug_group` and `pop_gl_debug_group` which push and pop respectively.